File: src/main/java/com/realtimeeventticketing/simulation/TicketEvent.java
package com.realtimeeventticketing.simulation;

public class TicketEvent {
    private final TicketEventType eventType;
    private final String message;
    private User user;
    private Ticket ticket;

    // Constructor
    public TicketEvent(TicketEventType eventType, String message) {
        this.eventType = eventType;
        this.message = message;
    }

    public TicketEvent(TicketEventType eventType, String message, User user, Ticket ticket) {
        this.eventType = eventType;
        this.message = message;
        this.user = user;
        this.ticket = ticket;
    }

    // Getters and Setters
    public TicketEventType getEventType() {
        return eventType;
    }

    public String getMessage() {
        return message;
    }

    public User getUser() {
        return user;
    }

    public Ticket getTicket() {
        return ticket;
    }
}

File: fe-client/src/app/models/TicketEvent.ts
import { Ticket } from './Ticket';
import {User} from './User';
import {Simulation} from './Simulation';

export enum EventType {
  TICKET_ADDED = 'TICKET_ADDED',
  TICKET_PURCHASED = 'TICKET_PURCHASED',
  POOL_FULL = 'POOL_FULL',
  POOL_EMPTY = 'POOL_EMPTY',
  SIMULATION_OVER = 'SIMULATION_OVER'
}

export interface TicketEvent {
  eventType: EventType;
  message: string;
  user?: User;
  ticket?: Ticket;
  simulationConfig: Simulation;
}
